pkg/asset/rhcos/bootstrap_image.go

```diff
@@ -0,0 +1,54 @@
+// Package rhcos contains assets for RHCOS.
+package rhcos
+
+import (
+	"context"
+	"time"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/rhcos"
+	"github.com/openshift/installer/pkg/types/baremetal"
+)
+
+// BootstrapImage is location of RHCOS image.
+// This stores the location of the image based on the platform.
+// eg. on AWS this contains ami-id, on Livirt this can be the URI for QEMU image etc.
+type BootstrapImage string
+
+var _ asset.Asset = (*BootstrapImage)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (i *BootstrapImage) Name() string {
+	return "BootstrapImage"
+}
+
+// Dependencies returns no dependencies.
+func (i *BootstrapImage) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate the RHCOS image location.
+func (i *BootstrapImage) Generate(p asset.Parents) error {
+	ic := &installconfig.InstallConfig{}
+	p.Get(ic)
+	config := ic.Config
+
+	var osimage string
+	var err error
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+	switch config.Platform.Name() {
+	case baremetal.Name:
+		osimage, err = rhcos.QEMU(ctx)
```

Comment:
Nit, but I think this could use a comment either here explaining why baremetal needs the QEMU image for bootstrap, or maybe a comment towards the top of the file explaining why this is separate.

For future visitors, they might be confused since baremetal IPI is the only case it's different.

Reply:
Done